analyzers/tests/SonarAnalyzer.UnitTest/Rules/SingleStatementPerLineTest.cs

```diff
@@ -40,6 +42,21 @@ public class SingleStatementPerLineTest
                 .WithTopLevelStatements()
                 .Verify();
 
+        [TestMethod]
+        public void SingleStatementPerLine_Razor() =>
+            builderCS.AddSnippet(
+"""
+@if (true) { @if (true) { <p>Test</p> }} <!-- FN -->
```

Comment:
Please add 2 separate cases on 2 separate lines, One with only C# statements and the other with the HTML mixed in. 
They might be fixed separately.

Reply:
I think you don't need to ifs per line. 1 should be enough.